source/framework/analysis/inc/TRestCalibrationCorrection.h

```diff
@@ -0,0 +1,224 @@
+        double GetIntercept(const double x, const double y) const;
+
+        Int_t GetPlaneId() const { return fPlaneId; }
+        Int_t GetModuleId() const { return fModuleId; }
+        std::string GetObservable() const { return p->fObservable; }
+        std::string GetSpatialObservableX() const { return p->fSpatialObservableX; }
+        std::string GetSpatialObservableY() const { return p->fSpatialObservableY; }
+        inline std::string GetModuleDefinitionCut() const { return fDefinitionCut; }
+        Int_t GetNumberOfSegmentsX() const { return fNumberOfSegmentsX; }
+        Int_t GetNumberOfSegmentsY() const { return fNumberOfSegmentsY; }
+
+        std::set<double> GetSplitX() const { return fSplitX; }
+        std::set<double> GetSplitY() const { return fSplitY; }
+        std::string GetDataSetFileName() const { return fDataSetFileName; }
+        TVector2 GetReadoutRangeVar() const { return fReadoutRange; }
+
+        void DrawSpectrum();
+        void DrawSpectrum(const double x, const double y, TCanvas* c = nullptr);
+        void DrawSpectrum(const int index_x, const int index_y, TCanvas* c = nullptr);
+        void DrawFullSpectrum();
+
+        void DrawLinearFit();
+        void DrawLinearFit(const double x, const double y, TCanvas* c = nullptr);
+        void DrawLinearFit(const int index_x, const int index_y, TCanvas* c = nullptr);
+
+        void DrawGainMap(const int peakNumber = 0);
+
+        void Refit(const double x, const double y, const double energy, const TVector2& range);
+        void Refit(const int x, const int y, const int peakNumber, const TVector2& range);
+
+        void SetPlaneId(const Int_t& planeId) { fPlaneId = planeId; }
+        void SetModuleId(const Int_t& moduleId) { fModuleId = moduleId; }
+        void SetModuleDefinitionCut(const std::string& moduleDefinitionCut) {
+            fDefinitionCut = moduleDefinitionCut;
+        }
+        void SetCalibrationRange(const TVector2& calibrationRange) { fCalibRange = calibrationRange; }
+        void SetNBins(const Int_t& nBins) { fNBins = nBins; }
+
+        void SetNumberOfSegmentsX(const Int_t& numberOfSegmentsX) {
+            fNumberOfSegmentsX = numberOfSegmentsX;
+            SetSplitX();
+        }
+        void SetNumberOfSegmentsY(const Int_t& numberOfSegmentsY) {
+            fNumberOfSegmentsY = numberOfSegmentsY;
+            SetSplitY();
+        }
+        // void SetInputFile( const std::string &inputFile) { fInputFile = inputFile;}
+        // void SetOutputFile( const std::string &outputFile) { fOutputFile = outputFile;}
+        void SetDataSetFileName(const std::string& dataSetFileName) { fDataSetFileName = dataSetFileName; }
+        void SetReadoutRange(const TVector2& readoutRange) { fReadoutRange = readoutRange; }
+        void SetZeroPoint(const bool& ZeroPoint) { fZeroPoint = ZeroPoint; }
+        void SetAutoRangePeaks(const bool& autoRangePeaks) { fAutoRangePeaks = autoRangePeaks; }
+
+        void Print();
```

Comment:
```suggestion
        void Print() const;
```

(should also update the implementation)

If possible all these should be const since they are read only functions (sometimes its not possible or easy to do).